nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
```

Comment:
Is this intended to be mutable? Who manages this file?

Reply:
The config file should be managed by AdGuard Home, see my response to dotlambda's review for my rationale. If this is the case should it be placed in `stateDir`?

Reply:
I have briefly read some documentation from upstream and I understand now. This application is definitely an example of an application that should manage its own configuration... **not** NixOS, as you mention.

I would recommend not including this as an option so. You could probably do something really slick with `DynamicUser`, `RuntimeDirectory` (for the `pid` file and `reload`s) and `StateDirectory` as well...

Reply:
`DynamicUser` works! What do you think should remain configurable when using it? In my view `host` and `port` are the essentials, `pidFile` could just be set to `/run/AdGuardHome/AdGuardHome.pid` (I can't see a strong case as to the need to disable it), and logs could always be to stdout?

There could be an `extraArgs` flag, allowing verbose logs if a user needs it and supplying an escape hatch for future command-line options.

Reply:
That sounds good 👍 Whatever you think is needed.